src/components/product/OutOfStock/OutOfStock.tsx

```diff
@@ -24,48 +26,94 @@ export interface OutOfStockProps {
   title?: string
   /**
    * Notification icon.
-   * @default <Icon name="Bell" />
+   * @default <Icon name="BellRinging" />
    */
   icon?: ReactElement
+  /**
+   * Notification icon.
+   * @default <Icon name="BellRinging" />
+   */
+  notificationIcon?: ReactElement
   /**
    *
    */
   onSubmit: (value: string) => void
 }
 
 function OutOfStock(props: OutOfStockProps) {
+  const defaultButtonTxt = 'Notify me'
+  const defaultIcon = (
+    <Icon name="BellRinging" weight="bold" width={16} height={16} />
+  )
+
+  const [btTxt, setBtTxt] = useState(defaultButtonTxt)
+  const [disabled, setDisabled] = useState(false)
+  const [iconButton, setIconButton] = useState(defaultIcon)
+  const [email, setEmail] = useState('')
+
   const {
     title = 'Out of Stock',
     notificationMsg = 'Notify me when available',
-    buttonTxt = 'Send',
-    icon = <Icon name="Bell" />,
+    buttonTxt = btTxt,
+    icon = iconButton,
+    notificationIcon = defaultIcon,
     onSubmit,
     testId = 'store-out-of-stock',
   } = props
 
-  const [email, setEmail] = useState('')
+  const reset = () => {
+    setIconButton(defaultIcon)
+    setBtTxt(defaultButtonTxt)
+    setDisabled(false)
+
+    setEmail('')
+  }
+
+  const handleSubmit = async () => {
+    setDisabled(true)
+    setIconButton(<Icon name="Ellipsis" weight="bold" width={16} height={16} />)
+
+    try {
+      onSubmit(email)
+      setIconButton(
+        <Icon name="Checked" weight="bold" width={16} height={16} />
+      )
+      setBtTxt('Subscribed successfully')
+
+      // Return to original state after 2s
+      await new Promise((r) => setTimeout(r, 2000)).then(reset)
```

Comment:
This fake request/forced timeout looks good for the demo but I'm not sure we should keep it here.

Reply:
Feel skeptical about that too, but I'm not sure what would be alternative for that.